app/src/Model/Entity/Article.php

```diff
@@ -0,0 +1,60 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Model\Entity;
+
+use Cake\ORM\Entity;
+use Cake\Collection\Collection;
+
+/**
+ * Article Entity
+ *
+ * @property int $id
+ * @property int $user_id
+ * @property string $title
+ * @property string $slug
+ * @property string $body
+ * @property bool $published
+ * @property \Cake\I18n\FrozenTime $created
+ * @property \Cake\I18n\FrozenTime $modified
+ *
+ * @property \App\Model\Entity\Tag[] $tags
+ */
+class Article extends Entity
+{
+    /**
+     * Fields that can be mass assigned using newEntity() or patchEntity().
+     *
+     * Note that when '*' is set to true, this allows all unspecified fields to
+     * be mass assigned. For security purposes, it is advised to set '*' to false
+     * (or remove it), and explicitly make individual fields accessible as needed.
+     *
+     * @var array<string, bool>
+     */
+    protected $_accessible = [
+        'user_id' => true,
+        'title' => true,
+        'slug' => true,
+        'body' => true,
+        'published' => true,
+        'created' => true,
+        'modified' => true,
+        'tags' => true,
+        'tag_string' => true,
+    ];
+
+    protected function _getTagString()
+    {
+        if (isset($this->_fields['tag_string'])) {
+            return $this->_fields['tag_string'];
+        }
+        if (empty($this->tags)) {
+            return '';
+        }
+        $tags = new Collection($this->tags);
+        $str = $tags->reduce(function ($string, $tag) {
+            return $string . $tag->title . ', ';
+        }, '');
+        return trim($str, ', ');
```

Comment:
```suggestion
        }, '');

        return trim($str, ', ');
```

If you run the command `docker compose exec php-app composer cs-check`, you'll notice there's error for this line of code (and mores for other lines of code 😅)

For this line of code, we need to add 1 blank line before the return statement.

Reply:
Please run the command `docker compose exec php-app composer cs-check` and check the errors.

Reply:
Additionally, we've a command `docker compose exec php-app composer cs-fix` that is able to fix certain violations automatically (not all violations). Please give a try.

Reply:
tried and got an error: anyway solve composer cs-check errors are easy to solve